Error compiled with librespot

I've been trying to fill it out, from now on, to the source of librespot.

But I get this error, I was really interested in solving the gapless problem:



Compiling librespot-protocol v0.1.1 (/tmp/librespot/protocol)
   Compiling librespot-core v0.1.1 (/tmp/librespot/core)
   Compiling librespot-audio v0.1.1 (/tmp/librespot/audio)
   Compiling librespot-metadata v0.1.1 (/tmp/librespot/metadata)
   Compiling librespot-playback v0.1.1 (/tmp/librespot/playback)
warning: unnecessary parentheses around type
  --> /tmp/librespot/playback/src/audio_backend/alsa.rs:28:42
   |
28 | fn open_device(dev_name: &str) -> Result<(PCM), Box<Error>> {
   |                                          ^^^^^ help: remove these parentheses
   |
   = note: `#[warn(unused_parens)]` on by default

warning: unnecessary parentheses around type
  --> /tmp/librespot/playback/src/mixer/alsamixer.rs:13:61
   |
13 |     fn map_volume(&self, set_volume: Option<u16>) -> Result<(u16), Box<dyn Error>> {
   |                                                             ^^^^^ help: remove these parentheses

   Compiling librespot-connect v0.1.1 (/tmp/librespot/connect)
   Compiling librespot v0.1.1 (/tmp/librespot)
   Compiling spotifyd v0.2.24 (/tmp/spotifyd)
**error[E0063]: missing field `gapless` in initializer of `librespot_playback::config::PlayerConfig`**
   --> src/config.rs:684:24
    |
684 |         player_config: PlayerConfig {
    |                        ^^^^^^^^^^^^ missing `gapless`

error[E0027]: pattern does not mention fields `play_request_id`, `position_ms`
  --> src/process.rs:67:9
   |
67 |         PlayerEvent::Started { track_id } => {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing fields `play_request_id`, `position_ms`

error[E0027]: pattern does not mention field `play_request_id`
  --> src/process.rs:71:9
   |
71 |         PlayerEvent::Stopped { track_id } => {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing field `play_request_id`

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0027, E0063.
For more information about an error, try `rustc --explain E0027`.
error: could not compile `spotifyd`.

To learn more, run the command again with --verbose.

I'm also having a problem with it.
Version 0.1.1 seems to generally be problematic, as it doesn't even compile its doc or shows the source files in the crate repo
https://docs.rs/crate/librespot-protocol/0.1.1

I'm getting the following error when trying to compile

```
error[E0308]: mismatched types
  --> /home/ido/.cargo/registry/src/github.com-1ecc6299db9ec823/librespot-protocol-0.1.1/build.rs:45:39
   |
45 |         let p = parse_and_typecheck(&["proto"], &[src]).expect("protoc");
   |                                       ^^^^^^^ expected struct `std::path::Path`, found `str`
   |
   = note: expected reference `&std::path::Path`
              found reference `&'static str`

error[E0308]: mismatched types
  --> /home/ido/.cargo/registry/src/github.com-1ecc6299db9ec823/librespot-protocol-0.1.1/build.rs:45:51
   |
45 |         let p = parse_and_typecheck(&["proto"], &[src]).expect("protoc");
   |                                                   ^^^ expected struct `std::path::Path`, found struct `std::string::Str  ing`
   |
   = note: expected reference `&std::path::Path`
              found reference `&std::string::String`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `librespot-protocol`.

```

--- a/README.md
+++ b/README.md
@@ -64,6 +64,16 @@ To install the resulting binary, run
 cargo install --path . --locked
 ```
 
+### Installing with Cargo
+
+If you have `cargo` installed, you can directly install `spotifyd` by running:
+
+```bash
+cargo install spotifyd --locked
+```
+
+That will compile and install `spotifyd`'s latest version under `$HOME/.cargo/bin` for you.
+
 #### Building a Debian package
 
 You can use the `cargo-deb` create in order to build a Debian package from source.

